src/epcpm/staticmodbusmodel.py

```diff
@@ -0,0 +1,1356 @@
+#                     marshmallow.fields.Nested(
+#                         graham.schema(TableRepeatingBlockReference)
+#                     ),
+#                 )
+#             ),
+#         ),
+#     )
+#
+#     uuid = epyqlib.attrsmodel.attr_uuid()
+#
+#     def __attrs_post_init__(self):
+#         super().__init__()
+#
+#     def remove_old_on_drop(self, node):
+#         return False
+#
+#     def child_from(self, node):
+#         reference = TableRepeatingBlockReference(original=node)
+#         for child in node.tree_parent.children:
+#             if not isinstance(child, DataPoint):
+#                 continue
+#
+#             reference.append_child(
+#                 TableRepeatingBlockReferenceDataPointReference(
+#                     original=child,
+#                 ),
+#             )
+#
+#         return reference
+#
+#     def can_drop_on(self, node):
+#         return isinstance(node, TableRepeatingBlock) and not any(
+#             isinstance(child, TableRepeatingBlockReference) for child in self.children
+#         )
+#
+#     def can_delete(self, node=None):
+#         if node is None:
+#             return self.tree_parent.can_delete(node=self)
+#
+#         return not isinstance(node, (HeaderBlock, FixedBlock))
+#
+#     def check_offsets_and_length(self):
+#         length = 0
+#
+#         for block in self.children:
+#             length += block.check_offsets_and_length()
+#
+#         return length
+#
+#     internal_move = epyqlib.attrsmodel.default_internal_move
+#     check = epyqlib.attrsmodel.check_just_children
+
+
+# class Repeating...?
+
+
+Root = epyqlib.attrsmodel.Root(
+    default_name="Static Modbus",
+    valid_types=(
+        DataPoint,
+        Table,
+        TableRepeatingBlockReferenceDataPointReference,
+        DataPointBitfield,
+        DataPointBitfieldMember,
+    ),
+    # valid_types=(Model, Table),
+)
+
+
+types = epyqlib.attrsmodel.Types(
+    types=(
+        Root,
+        # Model,
+        DataPoint,
+        DataPointBitfield,
+        DataPointBitfieldMember,
+        # HeaderBlock,
+        # FixedBlock,
+        Table,
+        TableRepeatingBlock,
+        TableRepeatingBlockReference,
+        TableRepeatingBlockReferenceDataPointReference,
+    ),
+)
+
+
+# TODO: CAMPid 943896754217967154269254167
+def merge(name, *types):
+    return tuple((x, name) for x in types)
+
+
+columns = epyqlib.attrsmodel.columns(
+    (
+        merge(
+            "name",
+            # HeaderBlock,
+            # FixedBlock,
+            Table,
+            TableRepeatingBlock,
+            TableRepeatingBlockReference,
+        )
+        # + merge("name", Model)
+        # + merge("id", Model)
+        + merge(
+            "parameter_uuid",
+            DataPoint,
+            TableRepeatingBlockReferenceDataPointReference,
+            DataPointBitfield,
+            DataPointBitfieldMember,
+        )
+    ),
+    merge("abbreviation", TableRepeatingBlock),
```

Comment:
We should talk about whether we want abbreviations in the new system.

Reply:
Commented out in ba23dc821ff9b6427c221e05870fcbf11ae43ac2